File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/approach/ServerSettingsMessage.scala





















// Generated on 05/03/2013 10:34:35
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ServerSettingsMessage extends MessageDeserializer {
  val messageId = 6340

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ServerSettingsMessage.messageId

    val lang: String
    val community: Byte

    def serialize(buf: Buffer) {
      buf.writeString(lang)
      buf.writeByte(community)
    }

  }

  protected def create(lang: String, community: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val lang = buf.readString()
    val community = buf.readByte()

    create(lang, community)
  }

}

object ServerSettingsMessage extends ServerSettingsMessage {
  case class MessageImpl(lang: String, community: Byte) extends Message
  type Target = MessageImpl

  override def create(lang: String, community: Byte) = MessageImpl(lang, community)

  def apply(lang: String, community: Byte) = create(lang, community)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ServerSettingsMessage.java


// Generated on 05/08/2013 19:37:40
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ServerSettingsMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6340;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public String lang;
    public byte community;
    public byte gameType;
    
    public ServerSettingsMessage() { }
    
    public ServerSettingsMessage(String lang, byte community, byte gameType) {
        this.lang = lang;
        this.community = community;
        this.gameType = gameType;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(lang);
        buf.writeByte(community);
        buf.writeByte(gameType);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        lang = buf.readString();
        community = buf.readByte();
        if (community < 0)
            throw new RuntimeException("Forbidden value on community = " + community + ", it doesn't respect the following condition : community < 0");
        gameType = buf.readByte();
        if (gameType < 0)
            throw new RuntimeException("Forbidden value on gameType = " + gameType + ", it doesn't respect the following condition : gameType < 0");
    }
    
}
